lib/FusionInventory/Agent/Task/Inventory.pm

```diff
@@ -248,6 +253,9 @@ sub _runModule {
         die "module $other_module, needed before $module, not found"
             if !$self->{modules}->{$other_module};
 
+        next if (!$self->{modules}->{$other_module}->{enabled} &&
+            $self->{modules}->{$module}->{runAfterIfEnabled}->{$other_module});
+
```

Comment:
I'd rather test the disabled condition once only, for better evidence:

# check the required module status
if (!$self->{modules}->{$other_module})->{enabled} {
    if ($self->{modules}->{$module}->{runAfterIfEnabled}->{$other_module}) {
        # soft dependency: run current module without required one
        next;
    } else {
        # hard dependency: abort current module execution
        die "module $other_module, needed before $module, not enabled"
    }
}